addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1038 @@
+            # should be only updated on the real ID and not on the virtual (like message_follower_ids):
+            # then set real ids to be updated.
+            if not values.get('recurrency', True) or not _only_changes_to_apply_on_real_ids(values.keys()):
+                ids.append(real_event_id)
+                continue
+            else:
+                event = self.browse(event_id)
+                if event.rrule:
+                    new_id = event._detach_one_event(values)
+
+        res = super(CalendarEvent, self.browse([int(event_id) for event_id in ids])).write(values)
+
+        # set end_date for calendar searching
+        record = self.browse(ids)
+        if values.get('recurrency', True) and values.get('end_type', 'count') in ('count', unicode('count')) and \
+                (values.get('rrule_type') or values.get('count') or values.get('start') or values.get('stop')):
+            for id in ids:
+                final_date = record._get_recurrency_end_date()
+                super(CalendarEvent, record).write({'final_date': final_date})
+
+        attendees_create = False
+        if values.get('partner_ids'):
+            attendees_create = record.create_attendees()
+
+        if (values.get('start_date') or values.get('start_datetime')) and values.get('active', True):
+            the_id = new_id or (ids and int(ids[0]))
+            if the_id:
+                if attendees_create:
+                    attendees_create = attendees_create[the_id]
+                    mail_to_ids = attendees_create['old_attendee_ids'] - attendees_create['removed_attendee_ids']
+                else:
+                    mail_to_ids = self.browse(the_id).attendee_ids
+
+                if mail_to_ids:
+                    mail_to_ids._send_mail_to_attendees(template_xmlid='calendar_template_meeting_changedate', email_from=self.env.user.email)
+        return bool(res)
+
+    @api.model
+    def create(self, vals):
+        self._set_date(vals)
+        res = super(CalendarEvent, self).create(vals)
+        final_date = res._get_recurrency_end_date()
+        res.write({'final_date': final_date})
+        res.create_attendees()
+        return res
+
+    def export_data(self, *args, **kwargs):
+        """ Override to convert virtual ids to ids """
+        return super(CalendarEvent, self.browse(list(set(get_real_ids(self.ids))))).export_data(*args, **kwargs)
+
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        if 'date' in groupby:
+            raise UserError(_('Group by date is not supported, use the calendar view instead.'))
+        return super(CalendarEvent, self.with_context({'virtual_id': False})).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
+        fields2 = fields or None
```

Comment:
Here you are assigning None as a fallback and at line you are appending 970, don't you think this can raise issue